src/js/helpers/ProjectOverwriteHelpers.js

```diff
@@ -135,7 +215,113 @@ export const createExternalVerseEdit = (projectPath, verseBefore, verseAfter, bo
       groupId: 'N/A'
     },
   };
-  const newFilename = modifiedTimestamp + '.json';
   const verseEditsPath = path.join(projectPath, '.apps', 'translationCore', 'checkData', 'verseEdits', bookId, chapter.toString(), verse.toString());
-  fs.outputJsonSync(path.join(verseEditsPath, newFilename.replace(/[:"]/g, '_')), verseEdit);
+  writeCheckData(verseEdit, verseEditsPath);
 };
+
+/**
+* @description gets the group data for the current verse from groupsDataReducer
+* @param {Object} groupsData
+* @param {Object} contextId
+* @return {object} group data object.
+*/
+export function getGroupDataForVerse(groupsData, contextId) {
+  const filteredGroupData = {};
+  for (let groupItemKey of Object.keys(groupsData)) {
+    const groupItem = groupsData[groupItemKey];
+    if (groupItem) {
+      for (let check of groupItem) {
+        try {
+          if (isEqual(check.contextId.reference, contextId.reference)) {
+            if (!filteredGroupData[groupItemKey]) {
+              filteredGroupData[groupItemKey] = [];
+            }
+            filteredGroupData[groupItemKey].push(check);
+          }
+        } catch (e) {
+          console.warn(`Corrupt check found in group "${groupItemKey}"`, check);
+        }
+      }
+    }
+  }
+  return filteredGroupData;
+}
+
+/**
+ * returns true if contextIds are a match for reference and group
+ * @param {Object} contextId1
+ * @param {Object} contextId2
+ * @return {boolean}
+ */
+export function sameContext(contextId1, contextId2) {
```

Comment:
Isnt this function a duplicate ?

Reply:
The function I took this from is inside the `checking-tool-wrapper`

Reply:
I just found it is also located in selections actions. I will update

Reply:
check SelectionsActions.js line 343